packages/core/src/App/Components/Elements/LandscapeBlocker/landscape-blocker.tsx

```diff
@@ -1,21 +1,29 @@
 import React from 'react';
 import { useLocation } from 'react-router-dom';
-import { isDisabledLandscapeBlockerRoute, isTabletOs, routes } from '@deriv/shared';
+import { isDisabledLandscapeBlockerRoute, isMobileOs, isTabletOs, routes } from '@deriv/shared';
 import { observer, useStore } from '@deriv/stores';
 import LandscapeBlockerSvg from 'Assets/SvgComponents/settings/landscape-blocker.svg';
 import './landscape-blocker.scss';
+import { useDevice } from '@deriv-com/ui';
 
 const LandscapeBlocker = observer(() => {
     // need to check for wallet account and don't hide landscape blocker for users migrated to wallets
     const {
         client: { has_wallet },
     } = useStore();
+    const { isMobile } = useDevice();
     const location = useLocation();
     const pathname = location?.pathname;
     const is_hidden_landscape_blocker = isDisabledLandscapeBlockerRoute(pathname);
     const shouldShowDtraderTabletView = pathname === routes.trade && isTabletOs;
+    const showBlockerDtraderMobileLandscapeView = !isMobile && isMobileOs() && pathname === routes.trade;
```

Comment:
Not sure what should be done here but I feel like this condition `!isMobile && isMobileOs()` contradicts itself

Reply:
no. isMobile is based on device dimension. isMobileOS is based on device meta data. I was confused at first myself

Reply:
isMobile is false when we switch from portrait to landscape

Reply:
So the logic goes like this. if the `device is landscape` and `device OS is mobile` and `pathname starts with trader` we show landscape blocker.